File: src/main/kotlin/wowsft/model/gameparams/ship/component/atba/Secondary.java
package wowsft.model.gameparams.ship.component.atba;

import wowsft.config.WoWSFT;
import wowsft.model.gameparams.TypeInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;

import java.util.List;

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
public class Secondary
{
    private List<String> ammoList;
    private float antiAirAuraDistance;
    private float antiAirAuraStrength;
    private float barrelDiameter;
    private List<List<Float>> deadZone;
    private float delim;
    private List<Float> horizSector;
    private long id;
    private float idealDistance;
    private float idealRadius;
    private String index;
    private float minRadius;
    private String name;
    private int numBarrels;
    private float radiusOnDelim;
    private float radiusOnMax;
    private float radiusOnZero;
    private List<Float> rotationSpeed;
    private float shotDelay;
    private boolean smallGun;
    private float smokePenalty;
    private TypeInfo typeinfo;
    private List<Float> vertSector;
    @JsonInclude
    private float GSIdealRadius = 1f;

    private int count;

    private float alphaDamage;
    private float alphaPiercingHE;
    private String ammoType;
    private float bulletSpeed;
    private float burnProb;

    public float getAlphaPiercingHEReal()
    {
        return alphaPiercingHE - 1f;
    }
}

File: src/main/kotlin/wowsft/model/gameparams/ship/component/atba/Secondary.kt
package wowsft.model.gameparams.ship.component.atba

import wowsft.config.WoWSFT
import wowsft.model.gameparams.TypeInfo
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
class Secondary {
    var ammoList = ArrayList<String>()
    var antiAirAuraDistance = 0.0
    var antiAirAuraStrength = 0.0
    var barrelDiameter = 0.0
    var deadZone = ArrayList<MutableList<Double>>()
    var delim = 0.0
    var horizSector = ArrayList<Double>()
    var id: Long = 0
    var idealDistance = 0.0
    var idealRadius = 0.0
    var index: String? = null
    var minRadius = 0.0
    var name: String? = null
    var numBarrels = 0
    var radiusOnDelim = 0.0
    var radiusOnMax = 0.0
    var radiusOnZero = 0.0
    var rotationSpeed = ArrayList<Double>()
    var shotDelay = 0.0
    var smallGun = false
    var smokePenalty = 0.0
    var typeinfo : TypeInfo? = null
    var vertSector = ArrayList<Double>()
    @JsonInclude
    var GSIdealRadius = 1.0

    var count = 0

    var alphaDamage = 0.0
    var alphaPiercingHE = 0.0
    var ammoType: String? = null
    var bulletSpeed = 0.0
    var burnProb = 0.0

    val alphaPiercingHEReal
        get() = alphaPiercingHE - 1.0
}
